pkg/kubelet/cm/internal_container_lifecycle.go

```diff
@@ -54,19 +54,18 @@ func (i *internalContainerLifecycleImpl) PreStartContainer(pod *v1.Pod, containe
 }
 
 func (i *internalContainerLifecycleImpl) PreStopContainer(containerID string) error {
-	if i.cpuManager != nil {
-		return i.cpuManager.RemoveContainer(containerID)
-	}
+	// We can't safely call i.cpuManager.RemoveContainer(containerID)
+	// here. Regular containers could be in the process of restarting, and
+        // RemoveContainer() would remove any allocated exclusive CPUs that the
+        // container should have.
 	return nil
 }
 
 func (i *internalContainerLifecycleImpl) PostStopContainer(containerID string) error {
-	if i.cpuManager != nil {
-		err := i.cpuManager.RemoveContainer(containerID)
-		if err != nil {
-			return err
-		}
-	}
+	// We can't safely call i.cpuManager.RemoveContainer(containerID)
+	// here. Regular containers could be in the process of restarting, and
+        // RemoveContainer() would remove any allocated exclusive CPUs that the
```

Comment:
I wouldn't bother with this comment here. I know we used to have code here that called RemoveContainer(), but it's more confusing to see the comment out of context, than to not see it at all -- especially since there is no path for calling RemoveContainer() from any external hooks anymore.

I actually plan to go through the TopologyManager after this and remove it's callout as well -- thus removing the need for this hook injection altogether.